lib/src/messaging.dart

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2018, Anatoly Pulyaevskiy. All rights reserved. Use of this source code
+// is governed by a BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+import 'package:node_interop/util.dart';
+
+import 'bindings.dart' as js show Messaging;
+import 'bindings.dart'
+    show
+        FcmMessage,
+        MulticastMessage,
+        MessagingPayload,
+        MessagingOptions,
+        MessagingDevicesResponse,
+        MessagingConditionResponse,
+        MessagingDeviceGroupResponse,
+        MessagingTopicResponse,
+        MessagingTopicManagementResponse,
+        BatchResponse;
+
+export 'bindings.dart'
+    show
+        FcmMessage,
+        TopicMessage,
+        TokenMessage,
+        ConditionMessage,
+        MulticastMessage,
+        Notification,
+        WebpushNotification,
+        WebpushFcmOptions,
+        WebpushConfig,
+        FcmOptions,
+        MessagingPayload,
+        DataMessagePayload,
+        NotificationMessagePayload,
+        MessagingOptions,
+        MessagingDevicesResponse,
+        MessagingDeviceResult,
+        MessagingConditionResponse,
+        MessagingDeviceGroupResponse,
+        MessagingTopicResponse,
+        MessagingTopicManagementResponse,
+        AndroidConfig,
+        AndroidFcmOptions,
+        AndroidNotification,
+        ApnsConfig,
+        ApnsFcmOptions,
+        ApnsPayload,
+        Aps,
+        ApsAlert,
+        CriticalSound,
+        BatchResponse,
+        SendResponse;
+
+class Messaging {
+  Messaging(this.nativeInstance);
+
+  @protected
+  final js.Messaging nativeInstance;
+
+  /// Sends the given [message] via FCM.
+  /// 
+  /// Returns Promise<string> fulfilled with a unique message ID string after the 
```

Comment:
Here and below Promise needs to be replaced with Future, maybe even rephrase it a bit, e.g. "Returns Future which completes with a unique message id string after the message..."